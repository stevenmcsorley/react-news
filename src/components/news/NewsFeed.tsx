import React, { useEffect, useState } from "react";

import RepositoryFactory from "../../api/respositoryFactory";
const NewsApi: any = RepositoryFactory.get("rssToJson");

interface NewsFeed {
  items: Items[];
}
interface Items {
  author: string;
  categories: [];
  content: string;
  description: string;
  enclosure?: [];
  guid: string;
  link: string;
  pubDate: string;
  thumbnail: string;
  title: string;
}

const NewsFeed = () => {
  const [query, setQuery] = useState<{ items: Items[] }>({
    items: [],
  });

  // const [isError, setIsError] = useState(false);
  // const [errorMsg, setErrorMsg] = useState(null);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    // setIsError(false);
    setIsLoading(true);

    const fetchData = async () => {
      const result = await NewsApi.getBreakingNews();
      console.log("result", result)
      setQuery(result.data);
      setIsLoading(false);
    };

    fetchData();
  }, []);


    return (
      <ul>
        {query.items.map((item, index) => !isLoading && <li>{item.title} - </li>)}
      </ul>
    );

};

export default NewsFeed;
