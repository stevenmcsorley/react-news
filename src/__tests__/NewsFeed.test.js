import React from 'react';
import { shallow } from 'enzyme';
import NewsFeed from '../components/news/NewsFeed'

test("testing NewsFeed feed",  async () =>{

    const NewsFeedComp = shallow(<NewsFeed />, 'div');
    expect(NewsFeedComp).toMatchSnapshot();

})