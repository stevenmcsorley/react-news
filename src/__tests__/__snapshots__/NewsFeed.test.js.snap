// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing NewsFeed feed 1`] = `ShallowWrapper {}`;
